### PR TITLE
build: setBuildEnv before building

### PR DESCRIPTION
**Pull request by oliverpool**
**What this PR does / why we need it**:

I am trying to cross-compile grafana using
```
go run build.go -goarch=armv6 -cgo-enabled=1 -cc=arm-linux-gnueabihf-gcc -pkg-arch=armhf build-server
```

It produces a binary, however running `file` on it yields:
```
grafana-server: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, BuildID[sha1]=b012673ccd9fadf7d426d646f1129836c69a7395, for GNU/Linux 3.2.0, not stripped
```

So I got a 64-bits x86-64 executable instead of a 32-bits armv6 :bug: 

I traced the issue in the `build` package, where the build is done before setting the `GO*` env variables.

**Special notes for your reviewer**:

Partial revert of 3cb8978f05a41348c786938ffd10caa49b6b50c0 (#38726):
https://github.com/grafana/grafana/pull/38726/files#diff-60d42249bb53df5d3ffd44f507aee87a57c2cff5a20f6b1d8403a76d0ce609b0L277-L287

### CODE REVIEW

**Comment by oliverpool** [comment]
As far as I know, this is still broken...

ping @kylebrandt @zserge

**Comment by oliverpool** [comment]
As far as I know, this is still broken...

ping @kylebrandt @zserge

**Comment by oliverpool** [comment]
As far as I know, this is still broken...

ping @kylebrandt @zserge

**Comment by oliverpool** [comment]
Very sad that there was no reaction whatsoever from the maintainers…

**Comment by dsotirakis** [comment]
@grafana/backend-platform looping you in!

**Review by zserge — APPROVED** [review]

**Comment by dsotirakis** [comment]
@oliverpool 👋 
Sorry for not taking care of this quicker. Could you please sign the CLA before we merge?

Thanks!

**Comment by oliverpool** [comment]
> Could you please sign the CLA before we merge?

I really like this project.
However I am sorry, but I can't sign your CLA.

A couple of reasons:
- I already signed a CLA with Grafana Labs: https://github.com/grafana/piechart-panel/pull/58#issuecomment-272832176 (even if this PR was also completely ignored by the maintainers).
- once signed, there is no way to end the CLA: https://github.com/cla-assistant/cla-assistant/issues/18 ("revoke for future (not yet opened) PRs")
- "is for your protection as a Contributor as well as the protection of Grafana Labs and its users": actually it is only for the benefit (not even "protection" of Grafana Labs, since a "developer certificate of origin" would provide such protection)

More discussions/context:
 - https://github.com/hasura/graphql-engine/pull/5323#issuecomment-655098038
 - https://drewdevault.com/2018/10/05/Dont-sign-a-CLA.html - https://drewdevault.com/2021/04/12/DCO.html

**Pull request closed without merging**
